### PR TITLE
return empty data frame (instead of NULL) for zero-row result

### PR DESCRIPTION
**Pull request by krlmlr**
implemented by generating a one-row data frame full of NA values and then chopping off that one row

### CODE REVIEW

**Comment by craigcitro** [comment]
LGTM -- is this easy enough to add a test for?

**Comment by krlmlr** [comment]
Will be tested implicitly with one of the DBItest tests. I'm not looking into Travis testing just yet -- this requires setting up an OAuth token that can be used by Travis (#14).

**Comment by krlmlr** [comment]
Merged with master.

File: R/tabledata.r [hunk 1]
```diff
@@ -144,7 +144,13 @@ converter <- list(
 )
 
 extract_data <- function(rows, schema) {
-  if (is.null(rows)) return(NULL)
+  if (is.null(rows)) {
+    # Corner case: Zero rows
+    dummy_rows <- list(list(f = rep(list(NULL), length(schema$fields))))
+
+    data <- extract_data(dummy_rows, schema)
+    return(data[integer(), , drop = FALSE])
```
Comment (hadley) [inline R/tabledata.r]: I think `data[0, , drop = FALSE]` is a more idiomatic way of doing this

**Pull request merged**